lib/coordinates.rb

```diff
@@ -0,0 +1,19 @@
+class Coordinates
+  attr_reader :value
+
+  def self.valid_values
+    %w[A1 A2 A3 B1 B2 B3 C1 C2 C3]
+  end
+
+  def initialize(value:)
+    @value = value
+  end
+
+  def valid?
+    self.class.valid_values.include?(value)
```

Comment:
I would typically make `self.class.valid_values` into an instance method that mirrors the class method. That would make this a lot easier to read: `valid_values.include?(value)`